viewing_party/party.py

```diff
@@ -0,0 +1,287 @@
+
+    # Watchlist is a list with multiple dictionaries
+    # Loop through each element in the watchlist list
+    # Check whether the title key matches the movie_title given above
+
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            watched_movie = movie
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(watched_movie)
+
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    """
+    input: A dictionary of user_data
+    output: The average rating of each movie in the user's watched list
+    """
+
+    # Access the watched key in user_data
+    # That points to a list
+    # Inside this list are dictionary elements
+    # We need to loop through each dictionary element
+    # in watched, access the rating keys
+    # store them in a variable for calculation
+    # calculate the average of each
+
+    sum = 0
+    avg_list = []
+
+    # store each rating int into a new list
+
+    if not user_data["watched"]:
+        return 0.0
+
+    else:
+        for movie in user_data["watched"]:
+            avg_list.append(movie["rating"])
+
+        for num in avg_list:
+            sum += num
+        
+        avg = sum / len(avg_list)
+        
+        return avg
+
+def get_most_watched_genre(user_data):
+    """
+    input: A dictionary of user_data
+    output: A string which represents which genre a user has watched the most
+    """
+    
+    # Given a dictionary user_data
+    # Access the watched key, which is a list
+    # Loop through that watched list
+    # Append each genre to a genre dictionary
+    # Each genre will equal a key
+    # Set the count of each genre equal to the value of its corresponding key
+    # Return the name of the most-watched genre
+
+    genre_list = []
+    genre_dict = {}
+
+    if not user_data["watched"]:
+        return None
+
+    else:
+        for movie in user_data["watched"]:
+            genre_list.append(movie["genre"])
+
+        # Search through genre_list
+        # Count how many times each genre is listed
+        # Add it to a dictionary
+
+        # Add each genre as a new key in the dictionary with an initalized value of 0
+        for item in genre_list:
+            genre_dict[item] = 0
+        # Go through each genre in genre_list, and count how many times it's in there
+        # Add this count to the corresponding dictionary key in genre_dict
+        for item in genre_list:
+            if item in genre_dict.keys():
+                genre_dict[item] += 1
+
+        # Store the most_watched genre in a variable
+        # Use the max function to pass two arguments
+        # The first argument is the iterable (dictionary) to check
+        # The second argument accesses the key in genre_dict that has the highest value
+        # The .get function reference accesses the value of a particular key
+        most_watched = max(genre_dict, key=genre_dict.get)
+
+        print(most_watched)
```

Comment:
It’s best practice remove "debugging" print statements like this before committing and/or submitting.